### PR TITLE
bugfix: Try to fix flaky ScalaCLI test

### PR DESCRIPTION
**Pull request by tgodzik**
It seems that there was a race condition between when `.bsp` directory was created and the actual json creation.

Now we wait until the json file is created

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 1]
```diff
@@ -1410,9 +1410,14 @@ class MetalsLanguageServer(
    */
   private def fileWatchFilter(path: Path): Boolean = {
     val abs = AbsolutePath(path)
-    abs.isScalaOrJava || abs.isSemanticdb || abs.isBuild || abs.isBsp
```
Comment (tanishiking) [inline metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:1413]: [nit]
Looks like `isBsp` is no longer used and we can remove :)

File: metals/src/main/scala/scala/meta/internal/metals/watcher/FileWatcher.scala [hunk 2]
```diff
@@ -51,12 +51,12 @@ final class FileWatcher(
...
 
   def start(
-      files: Set[AbsolutePath]
+      directories: Set[AbsolutePath]
   ): Unit = {
     stopWatcher = startWatch(
       config,
       workspaceDeferred().toNIO,
-      PathsToWatch(files.map(_.toNIO), Set.empty),
+      PathsToWatch(Set.empty, directories.map(_.toNIO)),
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/watcher/FileWatcher.scala]: Good idea! I was not thinking about generalizing it.

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala [hunk 3]
```diff
@@ -1410,9 +1410,14 @@ class MetalsLanguageServer(
...
+    abs.isScalaOrJava || abs.isSemanticdb || abs.isBuild || isInBspDirectory(
+      abs
+    )
   }
 
+  private def isInBspDirectory(path: AbsolutePath) =
+    path.toNIO.startsWith(workspace.resolve(".bsp").toNIO)
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala]: Moved!

File: metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala (hunk 1 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala:1413]: Removed

**Pull request merged**